pymodm/dereference.py

```diff
@@ -131,7 +131,7 @@ def _attach_objects_in_path(container, document_map, fields, key, field):
         for idx, item in enumerate(value):
             _attach_objects_in_path(value, document_map, fields,
                                     idx, field._field)
-    elif isinstance(field, EmbeddedDocumentListField):
+    elif isinstance(field, EmbeddedModelListField):
```

Comment:
This should use `_EmbeddedDocumentListFieldBase` so that it works with both the new and old classes.